### DISCUSSION TITLE
Can CroppedBitmap be saved as a local image?

### DISCUSSION THREAD

**Discussion by nir34** [AvaloniaUI/Avalonia#13772] [Q&A] [answered]
CroppedBitmap does not inherit from the bitmap class and has no save() method. I tried cropping the bitmap using writeableBitmap.CopyPixels(sourceRect, fb.Address, fb.RowBytes * writeableBitmap.PixelSize.Height, fb.RowBytes) but the result is not as good as I would have liked, it seems that the The x,y values of sourceRect don't work, only the CroppedBitmap can be cropped properly.

**Comment by timunie** [comment]
Ah looks like this is a "control" rather than a bitmap. So maybe [#13358](https://github.com/AvaloniaUI/Avalonia/discussions/13358#discussioncomment-7365561) is a solution for you.

**Comment by nir34** [comment, marked as answer]
` var image = croppedBitmap.Source as WriteableBitmap;
  Stream xxStream = new MemoryStream();
 image.Save(xxStream);
 xxStream.Seek(0, SeekOrigin.Begin);
 var skBitmapSource = SKBitmap.Decode(xxStream);

 using var skBitmapCrop = new SKBitmap(croppedBitmap.SourceRect.Width, croppedBitmap.SourceRect.Height);
 using (var skCanvas = new SKCanvas(skBitmapCrop))
 {
     var source = new SKRect(croppedBitmap.SourceRect.X, croppedBitmap.SourceRect.Y, croppedBitmap.SourceRect.Width+ croppedBitmap.SourceRect.X, croppedBitmap.SourceRect.Height+ croppedBitmap.SourceRect.Y);
     var dest = new SKRect(0, 0, croppedBitmap.SourceRect.Width , croppedBitmap.SourceRect.Height);

     skCanvas.DrawBitmap(skBitmapSource, source, dest);
 }

 var skBitmap = skBitmapCrop;`
